volatility3/framework/plugins/windows/amcache.py

```diff
@@ -0,0 +1,603 @@
+            if isinstance(last_mod_2, datetime.datetime):
+                yield f"Amcache: {entry_type} {path} STANDARD_INFORMATION create time", timeliner.TimeLinerType.CREATED, last_mod_2
+            if isinstance(install_time, datetime.datetime):
+                yield f"Amcache: {entry_type} {path} installed", timeliner.TimeLinerType.CREATED, install_time
+            if isinstance(compile_time, datetime.datetime):
+                yield f"Amcache: {entry_type} {path} compiled (PE metadata)", timeliner.TimeLinerType.MODIFIED, compile_time
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        # Since we're calling the plugin, make sure we have the plugin's requirements
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.PluginRequirement(
+                name="hivelist", plugin=hivelist.HiveList, version=(1, 0, 0)
+            ),
+        ]
+
+    @classmethod
+    def get_amcache_hive(
+        cls,
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        kernel: interfaces.context.ModuleInterface,
+    ) -> Optional[registry.RegistryHive]:
+        """Retrieves the `Amcache.hve` registry hive from the kernel module, if it can be located."""
+        return next(
+            (
+                hive
+                for hive in hivelist.HiveList.list_hives(
+                    context=context,
+                    base_config_path=interfaces.configuration.path_join(
+                        config_path, "hivelist"
+                    ),
+                    layer_name=kernel.layer_name,
+                    symbol_table=kernel.symbol_table_name,
+                    filter_string="amcache",
+                )
+            ),
+            None,
+        )
+
+    @classmethod
+    def parse_file_key(
+        cls, file_key: reg_extensions.CM_KEY_NODE
+    ) -> Iterator[Tuple[str, AmcacheEntry]]:
+        """Parses File entries from the Windows 8 `Root\\File` key.
+
+        :param programs_key: The `Root\\File` registry key.
+
+        :return: An iterator of tuples, where the first member is the program ID string for
+        correlating `Root\\Program` entries, and the second member is the `AmcacheEntry`.
+        """
+        for file_entry_key in file_key.get_subkeys():
+            values = list(file_entry_key.get_values())
+
+            (
+                program_id,
+                path,
+                company,
+                product,
+                sha1_hash,
+                last_mod_raw,
+                last_mod_2_raw,
+                create_time_raw,
+            ) = (
+                _try_get_value(values, "100", reg_extensions.RegValueTypes.REG_SZ),
```

Comment:
Are these values and their meanings documented somewhere?  Might be nice to have an enum or a dictionary just for humans to translate what the number is to what the value means.  Just feels like like ordered lists of specific numbers like this are fragile, and keeping a separate clear table somewhere makes it easier to tinker or change without breaking everything like a giant game of jenga...  5;)

Reply:
I've created enums for each set of Amcache entry value value names, and added a reference in a comment for those who want to seek out more information